src/main/java/io/cdap/plugin/aws/sqs/source/SqsSourceUtil.java

```diff
@@ -0,0 +1,210 @@
+
+  private SqsSourceUtil() {
+    // no-op
+  }
+
+  static JavaDStream<StructuredRecord> getJavaDStream(StreamingContext context, SqsSourceConfig conf) {
+    return context.getSparkStreamingContext().receiverStream(SqsSourceUtil.getReceiver(conf));
+  }
+
+  /**
+   * Gets {@link JavaReceiverInputDStream} for {@link SqsSource}.
+   *
+   * @param conf {@link SqsSourceConfig} config
+   */
+  private static Receiver<StructuredRecord> getReceiver(SqsSourceConfig conf) {
+    return new Receiver<StructuredRecord>(StorageLevel.MEMORY_ONLY()) {
+      @Override
+      public StorageLevel storageLevel() {
+        return StorageLevel.MEMORY_ONLY();
+      }
+
+      @Override
+      public void onStart() {
+        new Thread() {
+          @Override
+          public void run() {
+            AmazonSQS sqs = null;
+            try {
+              sqs = SqsUtil.getSqsClient(conf.getAuthenticationMethod(), conf.getAccessId(),
+                conf.getAccessKey(), conf.getSqsEndpoint(), conf.getRegion().getName(), conf.getQueueName());
+            } catch (SqsInitializationException e) {
+              throw new RuntimeException(e);
+            }
+
+            while (!isStopped()) {
+              try {
+                List<Message> messages = fetchMessages(sqs, conf);
+                LOG.debug("messages.size() = {}", messages.size());
+                for (Message message : messages) {
+                  store(toRecord(message, conf));
+                }
+                if (conf.getDelete()) {
+                  deleteMessages(sqs, conf, messages);
+                }
+              } catch (Exception e) {
+                LOG.error(String.format("Error getting content from %s queue.", conf.getQueueName()), e);
+              }
+
+              if (conf.getInterval() > 0) {
+                try {
+                  TimeUnit.SECONDS.sleep(conf.getInterval());
+                } catch (InterruptedException e) {
+                  throw new RuntimeException(e);
+                }
+              }
+            }
+          }
+
+          @Override
+          public void interrupt() {
+            super.interrupt();
+          }
+        }.start();
+      }
+
+      @Override
+      public void onStop() {
+        // no-op
+      }
+
+      private List<Message> fetchMessages(AmazonSQS sqs, SqsSourceConfig config) throws Exception {
+        if (sqs == null) {
+          return Collections.emptyList();
+        }
+
+        String queueName = config.getQueueName();
+        String queueUrl = sqs.getQueueUrl(queueName).getQueueUrl();
+        // Enable long polling on an existing queue
+        SetQueueAttributesRequest setQueueAttributesRequest = new SetQueueAttributesRequest()
+          .withQueueUrl(queueUrl)
+          .addAttributesEntry("ReceiveMessageWaitTimeSeconds", String.valueOf(config.getWaitTime()));
+        sqs.setQueueAttributes(setQueueAttributesRequest);
```

Comment:
Does this have to be set every time we fetch messages or can it just be done when we create the client?

Reply:
Done